sdk/python/resources/connections/connections.ipynb

```diff
@@ -0,0 +1,217 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import WorkspaceConnection\n",
+    "from azure.ai.ml.entities import UsernamePasswordConfiguration\n",
+    "\n",
+    "credentials = UsernamePasswordConfiguration(\n",
+    "    username=\"<user_name_for_acr>\",\n",
+    "    password=\"<pwd_for_acr>\"\n",
+    ")\n",
+    "\n",
+    "ws_connection = WorkspaceConnection(\n",
+    "    name=\"<connection_name>\",\n",
+    "    target=\"<acr_url>\", # testacr.azurecr.io,\n",
+    "    type=\"container_registry\",\n",
+    "    credentials=credentials\n",
+    ")\n",
+    "\n",
+    "ml_client.connections.create_or_update(ws_connection)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2.2 Create a Workspace Connection for private feeds\n",
+    "In this example we will create a **Workspace Connection** to **private git feed** and use **Personal Access Token (PAT)** credentials\n",
+    "\n",
+    "Once connection is successfully created you can reference feeds from **private git feed** in AML environment and AML will automatically fetch these credentials while building the environment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import WorkspaceConnection\n",
+    "from azure.ai.ml.entities import PatTokenConfiguration\n",
+    "\n",
+    "credentials = PatTokenConfiguration(\n",
+    "    pat=\"<pat_to_git_repo>\"\n",
+    ")\n",
+    "\n",
+    "ws_connection = WorkspaceConnection(\n",
+    "    name=\"<connection_name>\",\n",
+    "    target=\"<git_url>\",\n",
+    "    type=\"git\",\n",
+    "    credentials=credentials\n",
+    ")\n",
+    "\n",
+    "ml_client.connections.create_or_update(ws_connection)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2.3 Create a Workspace Connection for private python feeds\n",
+    "In this example we will create a **Workspace Connection** to **private python feed** and use **SAS Token** credentials\n",
+    "\n",
+    "Once connection is successfully created you can reference feeds from **private python feed** in AML environment and AML will automatically fetch these credentials while building the environment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import WorkspaceConnection\n",
+    "from azure.ai.ml.entities import SasTokenConfiguration\n",
+    "\n",
+    "credentials = SasTokenConfiguration(\n",
+    "    pat=\"<sas_to_access_python_feed>\"\n",
```

Comment:
We should educate users not to hardcode passwords, but to use environment variables instead.

Reply:
Thanks @rastala  have addressed the comment